Svc/FpySequencer/FpySequencerDirectives.cpp

```diff
@@ -423,55 +426,72 @@ Signal FpySequencer::binaryCmp_directiveHandler(const FpySequencer_BinaryCmpDire
         return Signal::stmtResponse_success;
     }
 
-    // okay, it is an inequality comparison
-
-    // whether the comparison is signed or unsigned
-    bool sign = true;
-
-    if (directive.get_op() >= Fpy::DirectiveId::ULT && directive.get_op() <= Fpy::DirectiveId::UGE) {
-        sign = false;
-    }
-
     I8 cmpResult;
 
-    if (sign) {
-        cmpResult = (lhs == rhs) ? 0 : (lhs < rhs) ? -1 : 1;
+    if (directive.get_op() >= Fpy::DirectiveId::FEQ && directive.get_op() <= Fpy::DirectiveId::FGE) {
+        F64 frhs, flhs;
```

Comment:
This code in general would benefit from some helper functions.  e.g.  break out the float processing, the signed processing, and the unsigned processing from one another.